src/Nethermind/Nethermind.Network/P2P/RateLimitedPacketAcceptanceStrategy.cs

```diff
@@ -0,0 +1,44 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Network.Rlpx;
+
+namespace Nethermind.Network.P2P;
+
+public class RateLimitedPacketAcceptanceStrategy : IPacketAcceptanceStrategy
+{
+    private readonly int _byteLimit;
+    private readonly TimeSpan _throttle;
+
+    private int _bytesAccepted;
+    private DateTime _timestamp;
+
+    public RateLimitedPacketAcceptanceStrategy(int byteLimit, TimeSpan throttle)
+    {
+        _byteLimit = byteLimit;
+        _throttle = throttle;
+        _timestamp = DateTime.UnixEpoch;
+    }
+
+    public bool Accepts(ZeroPacket packet)
```

Comment:
You know, you could, simply replace `ZeroPacket` with `int bytesToAccept` and that would be enough to put this class in `Nethermind.Core`.